weather.py

```diff
@@ -90,7 +135,21 @@ def generate_summary(weather_data):
     Returns:
         A string containing the summary information.
     """
-    pass
+    quantity_of_days = len(weather_data)
+    list_of_dates = []
+    list_of_min_temp = []
+    list_of_max_temp = []
+    for item in weather_data:
+        list_of_dates.append(convert_date(item[0]))
+        list_of_min_temp.append(item[1])
+        list_of_max_temp.append(item[2])
+
+    lowest_temperature = find_min(list_of_min_temp)
+    highest_temperature = find_max(list_of_max_temp)
```

Comment:
This works well. Just a minor suggestion: since `find_min` returns a tuple, you can also use tuple unpacking to remove the need to use indexing, e.g.
```suggestion
    lowest_temperature, lowest_temperature_index = find_min(list_of_min_temp)
    highest_temperature, highest_temperature_index = find_max(list_of_max_temp)
```